spec/integration/app_spec.rb

```diff
@@ -129,6 +129,15 @@ def start_app(cmd, port, timeout)
           config_text = File.read(mongoid_config_file)
           config_text.should =~ /mongoid_test_config_development/
           config_text.should =~ /mongoid_test_config_test/
+
+          # deprecated options should not be included
+          config_text.should_not =~ /background_indexing/
+
+          # make sure the different option types are emitted
+          config_text.should =~ /# app_name: nil/
+          config_text.should =~ /# discriminator_key: "_type"/
+          config_text.should =~ /# join_contexts: false/
+          config_text.should =~ /# log_level: :info/
```

Comment:
I would be better to switch to expectation syntax everywhere:

```
expect(config_text).to match /.../
```

Note that .match? is faster than =~. (Mongoid would really benefit from Rubocop.)

Reply:
In this context, I'm not worried about optimizing for speed (I can't imagine `.match?` is fast enough here to make much difference), but it would be good to be consistent about which syntax we use in the tests. I'll update these to use expectation syntax.